lib/policyEvaluator/utils/actionMaps.js

```diff
@@ -0,0 +1,128 @@
+    bucketGet: 's3:ListBucket',
+    bucketGetACL: 's3:GetBucketAcl',
+    bucketGetCors: 's3:GetBucketCORS',
+    bucketGetLifecycle: 's3:GetLifecycleConfiguration',
+    bucketGetLocation: 's3:GetBucketLocation',
+    bucketGetReplication: 's3:GetReplicationConfiguration',
+    bucketGetVersioning: 's3:GetBucketVersioning',
+    bucketGetWebsite: 's3:GetBucketWebsite',
+    bucketHead: 's3:ListBucket',
+    bucketPutACL: 's3:PutBucketAcl',
+    bucketPutCors: 's3:PutBucketCORS',
+    bucketPutLifecycle: 's3:PutLifecycleConfiguration',
+    bucketPutReplication: 's3:PutReplicationConfiguration',
+    bucketPutVersioning: 's3:PutBucketVersioning',
+    bucketPutWebsite: 's3:PutBucketWebsite',
+    listMultipartUploads: 's3:ListBucketMultipartUploads',
+    listParts: 's3:ListMultipartUploadParts',
+    multipartDelete: 's3:AbortMultipartUpload',
+    objectDelete: 's3:DeleteObject',
+    objectDeleteTagging: 's3:DeleteObjectTagging',
+    objectGet: 's3:GetObject',
+    objectGetACL: 's3:GetObjectAcl',
+    objectGetTagging: 's3:GetObjectTagging',
+    objectPut: 's3:PutObject',
+    objectPutACL: 's3:PutObjectAcl',
+    objectPutTagging: 's3:PutObjectTagging',
+};
+
+// action map used for request context
+const actionMapRQ = Object.assign({
+    bucketPut: 's3:CreateBucket',
+    // for bucketDeleteCors need s3:PutBucketCORS permission
+    // see http://docs.aws.amazon.com/AmazonS3/latest/API/
+    // RESTBucketDELETEcors.html
+    bucketDeleteCors: 's3:PutBucketCORS',
+    bucketDeleteReplication: 's3:DeleteReplicationConfiguration',
+    bucketDeleteLifecycle: 's3:DeleteLifecycleConfiguration',
+    completeMultipartUpload: 's3:PutObject',
+    initiateMultipartUpload: 's3:PutObject',
+    objectDeleteVersion: 's3:DeleteObjectVersion',
+    objectDeleteTaggingVersion: 's3:DeleteObjectVersionTagging',
+    objectGetVersion: 's3:GetObjectVersion',
+    objectGetACLVersion: 's3:GetObjectVersionAcl',
+    objectGetTaggingVersion: 's3:GetObjectVersionTagging',
+    objectHead: 's3:GetObject',
+    objectPutACLVersion: 's3:PutObjectVersionAcl',
+    objectPutPart: 's3:PutObject',
+    objectPutTaggingVersion: 's3:PutObjectVersionTagging',
+    serviceGet: 's3:ListAllMyBuckets',
+    objectReplicate: 's3:ReplicateObject',
+}, sharedActionMap);
+
+// action map used for bucket policies
+const actionMapBP = Object.assign({
+    bucketDeletePolicy: 's3:DeleteBucketPolicy',
+    bucketGetObjectLock: 's3:GetBucketObjectLockConfiguration',
+    bucketGetPolicy: 's3:GetBucketPolicy',
+    bucketPutObjectLock: 's3:PutBucketObjectLockConfiguration',
+    bucketPutPolicy: 's3:PutBucketPolicy',
+    objectGetLegalHold: 's3:GetObjectLegalHold',
+    objectGetRetention: 's3:GetObjectRetention',
+    objectPutLegalHold: 's3:PutObjectLegalHold',
+    objectPutRetention: 's3:PutObjectRetention',
+}, sharedActionMap);
+
+// action map for all relevant s3 actions
+const actionMapS3 = Object.assign({
+    bucketGetNotification: 's3:GetBucketNotification',
```

Comment:
minor: you could have those two into its own `actionMapBucketNotifications` variable to make it clearer that S3 is an aggregate of multiple action sets.